src/borg/helpers.py

```diff
@@ -722,17 +722,7 @@ def format_archive(archive):
     )
 
 
-def memoize(function):
-    cache = {}
-
-    def decorated_function(*args):
-        try:
-            return cache[args]
-        except KeyError:
-            val = function(*args)
-            cache[args] = val
-            return val
-    return decorated_function
+memoize = lru_cache(maxsize=None)
```

Comment:
looks like it is only used in pwd context, so I'ld say replace them, but keep maxsize=None.